tests/acceptance/request_test.py

```diff
@@ -99,6 +114,7 @@ def test_400_if_required_body_is_missing(test_app):
     ).status_code == 400
 
 
+@pytest.mark.xfail
```

Comment:
Ah ya, this is an annoying part of the pyramid request object. If we can trust the content-type header, you could add a check for that in the PyramidRequest object, and not actually access the `POST` if the context is json.